graylog2-server/src/main/java/org/graylog/plugins/views/search/Search.java

```diff
@@ -168,12 +198,24 @@ public Set<String> usedStreamIds() {
         return Sets.union(queryStreamIds, searchTypeStreamIds);
     }
 
+    public Set<String> usedStreamCategories() {
+        return queries().stream()
+                .map(Query::usedStreamCategories)
+                .reduce(Collections.emptySet(), Sets::union);
+    }
+
     public Set<String> streamIdsForPermissionsCheck() {
         return queries().stream()
                 .map(Query::streamIdsForPermissionsCheck)
                 .reduce(Collections.emptySet(), Sets::union);
     }
 
+    public Set<String> streamCategoriesForPermissionsCheck() {
+        return queries().stream()
+                .map(Query::usedStreamCategories)
+                .reduce(Collections.emptySet(), Sets::union);
```

Comment:
If the implementation is the same as in `usedStreamCategories` method, maybe we should remove it?

Reply:
removed in latest commit